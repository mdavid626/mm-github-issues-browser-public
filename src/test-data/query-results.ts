export const issueQueryResult = {
  data: {
    repository: {
      id: 'MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==',
      issue: {
        id: 'I_kwDOAJy2Ks5ePiI3',
        createdAt: '2023-02-12T05:31:50Z',
        title: 'Bug: build args parse error on powsershell',
        state: 'CLOSED',
        number: 26156,
        stateReason: 'COMPLETED',
        url: 'https://github.com/facebook/react/issues/26156',
        body: 'test issue result',
        author: { login: 'weidehai', __typename: 'User' },
        comments: {
          nodes: [],
          pageInfo: {
            hasNextPage: false,
            endCursor: null,
            __typename: 'PageInfo',
          },
          totalCount: 0,
          __typename: 'IssueCommentConnection',
        },
        __typename: 'Issue',
      },
      __typename: 'Repository',
    },
  },
};
