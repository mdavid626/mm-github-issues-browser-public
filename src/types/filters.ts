export const stateFilters = ['open', 'closed'] as const;

export type StateFilter = (typeof stateFilters)[number];

export type Filters = {
  search: string;
  state: StateFilter | null;
};
