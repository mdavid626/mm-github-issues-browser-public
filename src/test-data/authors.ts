import { Author } from '../types/author';

export const testAuthor1: Author = {
  login: 'testAuthor1',
};
