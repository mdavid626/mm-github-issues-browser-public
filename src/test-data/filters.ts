import { Filters } from '../types/filters';

export const testFilters1: Filters = {
  search: 'test',
  state: 'open',
};
