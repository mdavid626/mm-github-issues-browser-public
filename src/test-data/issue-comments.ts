import { IssueComment, IssueComments } from '../types/issue-comment';
import { testAuthor1 } from './authors';

export const testIssueComment1: IssueComment = {
  id: 'IC_kwDOAJy2Ks5Ux_nZ',
  body: 'issue comment 1',
  createdAt: '2023-02-12T09:43:24.641Z',
  author: testAuthor1,
};

export const testIssueComment2: IssueComment = {
  id: 'IC_kwDOAJy2Ks5Ux4u7',
  body: 'issue comment 2',
  createdAt: '2022-01-11T10:42:25.655Z',
  author: testAuthor1,
};

export const testIssueComments1: IssueComments = {
  nodes: [testIssueComment1, testIssueComment2],
  pageInfo: {
    hasNextPage: true,
    endCursor: 'endCursor1',
  },
  totalCount: 2,
};

export const testEmptyIssueComments: IssueComments = {
  nodes: [],
  pageInfo: {
    hasNextPage: false,
    endCursor: 'endCursor1',
  },
  totalCount: 0,
};
