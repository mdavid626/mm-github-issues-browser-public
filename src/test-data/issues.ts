import { Issue, IssueQueryResult, IssuesItem } from '../types/issue';
import { testAuthor1 } from './authors';
import { testIssueComments1 } from './issue-comments';

export const testIssue1: Issue = {
  id: 'I_kwDOAJy2Ks5d65oG',
  createdAt: '2023-02-08T08:50:14Z',
  title: 'Bug: Failed when set homepage in package.json or set PUBLIC_URL',
  state: 'CLOSED',
  number: 26126,
  url: 'https://github.com/facebook/react/issues/26126',
  body: 'test issue1 body',
  author: testAuthor1,
  comments: testIssueComments1,
};

export const testIssuesItem1: IssuesItem = {
  id: 'I_kwDOAJy2Ks5d65oG',
  createdAt: '2023-02-08T08:50:14Z',
  title: 'Bug: Failed when set homepage in package.json or set PUBLIC_URL',
  state: 'CLOSED',
  number: 26126,
  url: 'https://github.com/facebook/react/issues/26126',
  author: testAuthor1,
};

export const testIssueQueryResult1: IssueQueryResult = {
  repository: {
    issue: testIssue1,
  },
};
