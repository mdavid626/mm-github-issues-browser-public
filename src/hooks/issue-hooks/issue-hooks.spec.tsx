import { act, waitFor } from '@testing-library/react';
import { testFilters1 } from '../../test-data/filters';
import {
  commentsQueryResult1,
  issueQueryResult1,
  issuesQueryResult1,
  loadMoreIssuesQueryResult1,
} from '../../test-data/query-results';
import {
  renderHookWithQueryClient,
  renderHookWithRouter,
} from '../../testing-library/render';
import {
  useFetchMoreIssueComment,
  useIssue,
  useIssueNumber,
  useIssues,
} from './issue-hooks';
import createSearchQueryText from './search-query-text';

jest.mock('./search-query-text');
describe('issue-hooks', () => {
  beforeEach(() => {
    fetchMock.resetMocks();
    jest.spyOn(window, 'alert').mockReturnValue(undefined);
    (createSearchQueryText as jest.Mock).mockReturnValue(
      'repo:facebook/react type:issue in:title test'
    );
  });
  afterEach(jest.resetAllMocks);

  describe('useIssueNumber', () => {
    it('should return issue number from URL', () => {
      const { result } = renderHookWithRouter(
        () => useIssueNumber(),
        undefined,
        ['/issue/123']
      );
      expect(result.current).toBe(123);
    });
  });

  describe('useIssue', () => {
    it('should load issue', async () => {
      fetchMock.mockResponse(JSON.stringify(issueQueryResult1));
      const { result } = renderHookWithQueryClient(() => useIssue(123));
      expect(result.current).toEqual([undefined, undefined]);
      await waitFor(() => {
        expect(result.current).toEqual([expect.any(Object), undefined]);
      });
      expect(result.current[0]).toMatchSnapshot('query result');
      expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(fetchMock.mock.calls[0]).toMatchSnapshot('query');
    });

    it('should return error when error', async () => {
      fetchMock.mockRejectedValue(new Error('test error'));
      const { result } = renderHookWithQueryClient(() => useIssue(123));
      expect(result.current).toEqual([undefined, undefined]);
      await waitFor(() => {
        expect(result.current).toEqual([undefined, expect.any(Error)]);
      });
    });
  });

  describe('useFetchMoreIssueComment', () => {
    it('should fetch more comments', async () => {
      fetchMock.mockResponseOnce(JSON.stringify(issueQueryResult1));
      fetchMock.mockResponseOnce(JSON.stringify(commentsQueryResult1));
      const { result } = renderHookWithQueryClient(() => {
        return [useIssue(123), useFetchMoreIssueComment(123)];
      });
      expect(result.current).toEqual([
        [undefined, undefined],
        [expect.any(Function), false],
      ]);
      await waitFor(() => {
        expect(result.current[0]).toEqual([expect.any(Object), undefined]);
      });
      expect(fetchMock).toHaveBeenCalledTimes(1);

      const [fetchMoreComments] = result.current[1] as ReturnType<
        typeof useFetchMoreIssueComment
      >;
      act(() => {
        fetchMoreComments('testCursor1');
      });
      await waitFor(() => {
        const [queryResult] = result.current[0] as ReturnType<typeof useIssue>;
        expect(queryResult?.repository.issue.comments.nodes).toHaveLength(5);
      });

      expect(fetchMock).toHaveBeenCalledTimes(2);
      expect(fetchMock.mock.calls[1]).toMatchSnapshot('query');
      expect(result.current[0][0]).toMatchSnapshot('issue');
    });

    it('should show error when error', async () => {
      fetchMock.mockRejectOnce(new Error('test error'));
      const { result } = renderHookWithQueryClient(() =>
        useFetchMoreIssueComment(123)
      );
      act(() => {
        result.current[0]('testCursor1');
      });
      await waitFor(() => {
        expect(window.alert).toHaveBeenCalledWith('test error');
      });
    });
  });

  describe('useIssues', () => {
    it('should load issues', async () => {
      fetchMock.mockResponse(JSON.stringify(issuesQueryResult1));
      const { result } = renderHookWithQueryClient(() =>
        useIssues(testFilters1)
      );
      expect(result.current).toEqual([
        undefined,
        true,
        undefined,
        expect.any(Function),
      ]);
      await waitFor(() => {
        expect(result.current).toEqual([
          expect.any(Object),
          false,
          undefined,
          expect.any(Function),
        ]);
      });
      expect(result.current[0]).toMatchSnapshot('query result');
      expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(fetchMock.mock.calls[0]).toMatchSnapshot('query');
      expect(createSearchQueryText).toHaveBeenCalledWith(testFilters1);
    });

    it('should return error when error', async () => {
      fetchMock.mockReject(new Error('test error'));
      const { result } = renderHookWithQueryClient(() =>
        useIssues(testFilters1)
      );
      expect(result.current).toEqual([
        undefined,
        true,
        undefined,
        expect.any(Function),
      ]);
      await waitFor(() => {
        expect(result.current[2]).toEqual(expect.any(Object));
      });
    });

    it('should load more issues', async () => {
      fetchMock.mockResponseOnce(JSON.stringify(issuesQueryResult1));
      fetchMock.mockResponseOnce(JSON.stringify(loadMoreIssuesQueryResult1));
      const { result } = renderHookWithQueryClient(() =>
        useIssues(testFilters1)
      );
      await waitFor(() => {
        expect(result.current[0]).toEqual(expect.any(Object));
      });
      result.current[3]();
      await waitFor(() => {
        expect(result.current[0]?.search.nodes).toHaveLength(4);
      });
      expect(fetchMock).toHaveBeenCalledTimes(2);
      expect(fetchMock.mock.calls[1]).toMatchSnapshot('query');
      expect(result.current[0]).toMatchSnapshot('query result');
    });
  });
});
