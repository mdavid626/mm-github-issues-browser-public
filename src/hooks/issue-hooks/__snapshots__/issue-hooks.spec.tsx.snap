// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-hooks useFetchMoreIssueComment should fetch more comments: issue 1`] = `
Object {
  "repository": Object {
    "__typename": "Repository",
    "id": "MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==",
    "issue": Object {
      "__typename": "Issue",
      "author": Object {
        "__typename": "User",
        "login": "ThanhPhucHuynh",
      },
      "body": "<!--
  Please provide a clear and concise description of what the bug is. Include
  screenshots if needed. Please test using the latest version of the relevant
  React packages to make sure your issue has not already been fixed.
-->
I want to use basePath in React like Nextjs: [Nextjs basePath](https://nextjs.org/docs/api-reference/next.config.js/basepath)
React version: 18.2
basePath:  test

## Steps To Reproduce

1. create react app and add homepage like 
\`\`\`
  .....
  \\"homepage\\": \\"/test\\",
  ....
\`\`\`
2. build and run with the command: \`serve -s build -l 44841 \`
3. access in the browser
<!--
  Your bug will get fixed much faster if we can run your code and it doesn't
  have dependencies other than React. Issues without reproduction steps or
  code examples may be immediately closed as not actionable.
-->

Link to code example: None

<!--
  Please provide a CodeSandbox (https://codesandbox.io/s/new), a link to a
  repository on GitHub, or provide a minimal code example that reproduces the
  problem. You may provide a screenshot of the application if you think it is
  relevant to your bug report. Here are some tips for providing a minimal
  example: https://stackoverflow.com/help/mcve.
-->

## The current behavior
![image](https://user-images.githubusercontent.com/48717211/217479112-4bfa388b-0ef0-4ea6-ad84-6c490ae4e98f.png)

## The expected behavior
It is okay when \`npm run start\`
but failed when built

Note: It's still okay when: \` \\"homepage\\": \\"http://localhost:45131\\",\`
not work:  \\"homepage\\": \\"http://localhost:45131/test\\" ",
      "comments": Object {
        "__typename": "IssueCommentConnection",
        "nodes": Array [
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "eps1lon",
            },
            "body": "Support requests filed as GitHub issues often go unanswered. We want you to find the answer you're looking for, so we suggest the following alternatives:

Coding Questions
If you have a coding question related to React and React DOM, it might be better suited for Stack Overflow. It's a great place to browse through frequent questions about using React, as well as ask for help with specific questions.

https://stackoverflow.com/questions/tagged/react

Talk to other React developers
There are many online forums which are a great place for discussion about best practices and application architecture as well as the future of React.

https://reactjs.org/community/support.html",
            "createdAt": "2023-02-08T10:52:54Z",
            "id": "IC_kwDOAJy2Ks5UyCVC",
          },
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "ThanhPhucHuynh",
            },
            "body": "I tried, just work in dev

On Wed, Feb 8, 2023, 17:21 Aravind D ***@***.***> wrote:

> how about building like this \\"homepage\\": \\"http://localhost:45131/test\\" ?
> Does that work?
>
> â€”
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/react/issues/26126#issuecomment-1422363579>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALTV3G6FCM4GWFEYVFXSXZLWWNXTPANCNFSM6AAAAAAUU66TOU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
",
            "createdAt": "2023-02-08T10:43:59Z",
            "id": "IC_kwDOAJy2Ks5Ux_nZ",
          },
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "arav-ind",
            },
            "body": "how about building like this \`\\"homepage\\": \\"http://localhost:45131/test\\"\` ? Does that work?
",
            "createdAt": "2023-02-08T10:21:30Z",
            "id": "IC_kwDOAJy2Ks5Ux4u7",
          },
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "ThanhPhucHuynh",
            },
            "body": "> Are you using react-router-dom? If yes, try adding the attribute basename='/test' to BrowserRouter.
> 
> \`\`\`
> <BrowserRouter basename='/test'>
>   <Routes>
>      <Route>
>   </Routes>
> </BrowserRouter>
> \`\`\`

I tried but it does not work. I follow [deploying React app into a subdirectory](https://skryvets.com/blog/2018/09/20/an-elegant-solution-of-deploying-react-app-into-a-subdirectory/)
not work when built and serve it",
            "createdAt": "2023-02-08T10:10:01Z",
            "id": "IC_kwDOAJy2Ks5Ux0zh",
          },
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "arav-ind",
            },
            "body": "Are you using react-router-dom? If yes, try adding the attribute basename",
            "createdAt": "2023-02-08T09:58:14Z",
            "id": "IC_kwDOAJy2Ks5Uxwq-",
          },
        ],
        "pageInfo": Object {
          "__typename": "PageInfo",
          "endCursor": "Y3Vyc29yOnYyOpK5MjAyMy0wMi0wOFQxMDo1ODoxNCswMTowMM5Uxwq-",
          "hasNextPage": false,
        },
        "totalCount": 5,
      },
      "createdAt": "2023-02-08T08:50:14Z",
      "id": "I_kwDOAJy2Ks5d65oG",
      "number": 26126,
      "state": "CLOSED",
      "stateReason": "NOT_PLANNED",
      "title": "Bug: Failed when set \\"homepage\\" in package.json or set PUBLIC_URL",
      "url": "https://github.com/facebook/react/issues/26126",
    },
  },
}
`;

exports[`issue-hooks useFetchMoreIssueComment should fetch more comments: query 1`] = `
Array [
  "https://api.github.com/graphql",
  Object {
    "body": "{\\"operationName\\":\\"getIssueComments\\",\\"variables\\":{\\"issueNumber\\":123,\\"commentsCursor\\":\\"testCursor1\\"},\\"query\\":\\"query getIssueComments($issueNumber: Int!, $commentsCursor: String) {\\\\n  repository(owner: \\\\\\"facebook\\\\\\", name: \\\\\\"react\\\\\\") {\\\\n    id\\\\n    issue(number: $issueNumber) {\\\\n      id\\\\n      comments(\\\\n        first: 3\\\\n        after: $commentsCursor\\\\n        orderBy: {field: UPDATED_AT, direction: DESC}\\\\n      ) {\\\\n        nodes {\\\\n          id\\\\n          createdAt\\\\n          body\\\\n          author {\\\\n            login\\\\n            __typename\\\\n          }\\\\n          __typename\\\\n        }\\\\n        pageInfo {\\\\n          hasNextPage\\\\n          endCursor\\\\n          __typename\\\\n        }\\\\n        totalCount\\\\n        __typename\\\\n      }\\\\n      __typename\\\\n    }\\\\n    __typename\\\\n  }\\\\n}\\"}",
    "headers": Object {
      "accept": "*/*",
      "content-type": "application/json",
    },
    "method": "POST",
    "signal": AbortSignal {},
  },
]
`;

exports[`issue-hooks useIssue should load issue: query 1`] = `
Array [
  "https://api.github.com/graphql",
  Object {
    "body": "{\\"operationName\\":\\"getIssue\\",\\"variables\\":{\\"issueNumber\\":123},\\"query\\":\\"query getIssue($issueNumber: Int!) {\\\\n  repository(owner: \\\\\\"facebook\\\\\\", name: \\\\\\"react\\\\\\") {\\\\n    id\\\\n    issue(number: $issueNumber) {\\\\n      id\\\\n      createdAt\\\\n      title\\\\n      state\\\\n      number\\\\n      stateReason\\\\n      url\\\\n      body\\\\n      author {\\\\n        login\\\\n        __typename\\\\n      }\\\\n      comments(first: 3, orderBy: {field: UPDATED_AT, direction: DESC}) {\\\\n        nodes {\\\\n          id\\\\n          createdAt\\\\n          body\\\\n          author {\\\\n            login\\\\n            __typename\\\\n          }\\\\n          __typename\\\\n        }\\\\n        pageInfo {\\\\n          hasNextPage\\\\n          endCursor\\\\n          __typename\\\\n        }\\\\n        totalCount\\\\n        __typename\\\\n      }\\\\n      __typename\\\\n    }\\\\n    __typename\\\\n  }\\\\n}\\"}",
    "headers": Object {
      "accept": "*/*",
      "content-type": "application/json",
    },
    "method": "POST",
    "signal": AbortSignal {},
  },
]
`;

exports[`issue-hooks useIssue should load issue: query result 1`] = `
Object {
  "repository": Object {
    "__typename": "Repository",
    "id": "MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==",
    "issue": Object {
      "__typename": "Issue",
      "author": Object {
        "__typename": "User",
        "login": "ThanhPhucHuynh",
      },
      "body": "<!--
  Please provide a clear and concise description of what the bug is. Include
  screenshots if needed. Please test using the latest version of the relevant
  React packages to make sure your issue has not already been fixed.
-->
I want to use basePath in React like Nextjs: [Nextjs basePath](https://nextjs.org/docs/api-reference/next.config.js/basepath)
React version: 18.2
basePath:  test

## Steps To Reproduce

1. create react app and add homepage like 
\`\`\`
  .....
  \\"homepage\\": \\"/test\\",
  ....
\`\`\`
2. build and run with the command: \`serve -s build -l 44841 \`
3. access in the browser
<!--
  Your bug will get fixed much faster if we can run your code and it doesn't
  have dependencies other than React. Issues without reproduction steps or
  code examples may be immediately closed as not actionable.
-->

Link to code example: None

<!--
  Please provide a CodeSandbox (https://codesandbox.io/s/new), a link to a
  repository on GitHub, or provide a minimal code example that reproduces the
  problem. You may provide a screenshot of the application if you think it is
  relevant to your bug report. Here are some tips for providing a minimal
  example: https://stackoverflow.com/help/mcve.
-->

## The current behavior
![image](https://user-images.githubusercontent.com/48717211/217479112-4bfa388b-0ef0-4ea6-ad84-6c490ae4e98f.png)

## The expected behavior
It is okay when \`npm run start\`
but failed when built

Note: It's still okay when: \` \\"homepage\\": \\"http://localhost:45131\\",\`
not work:  \\"homepage\\": \\"http://localhost:45131/test\\" ",
      "comments": Object {
        "__typename": "IssueCommentConnection",
        "nodes": Array [
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "eps1lon",
            },
            "body": "Support requests filed as GitHub issues often go unanswered. We want you to find the answer you're looking for, so we suggest the following alternatives:

Coding Questions
If you have a coding question related to React and React DOM, it might be better suited for Stack Overflow. It's a great place to browse through frequent questions about using React, as well as ask for help with specific questions.

https://stackoverflow.com/questions/tagged/react

Talk to other React developers
There are many online forums which are a great place for discussion about best practices and application architecture as well as the future of React.

https://reactjs.org/community/support.html",
            "createdAt": "2023-02-08T10:52:54Z",
            "id": "IC_kwDOAJy2Ks5UyCVC",
          },
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "ThanhPhucHuynh",
            },
            "body": "I tried, just work in dev

On Wed, Feb 8, 2023, 17:21 Aravind D ***@***.***> wrote:

> how about building like this \\"homepage\\": \\"http://localhost:45131/test\\" ?
> Does that work?
>
> â€”
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/react/issues/26126#issuecomment-1422363579>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALTV3G6FCM4GWFEYVFXSXZLWWNXTPANCNFSM6AAAAAAUU66TOU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
",
            "createdAt": "2023-02-08T10:43:59Z",
            "id": "IC_kwDOAJy2Ks5Ux_nZ",
          },
          Object {
            "__typename": "IssueComment",
            "author": Object {
              "__typename": "User",
              "login": "arav-ind",
            },
            "body": "how about building like this \`\\"homepage\\": \\"http://localhost:45131/test\\"\` ? Does that work?
",
            "createdAt": "2023-02-08T10:21:30Z",
            "id": "IC_kwDOAJy2Ks5Ux4u7",
          },
        ],
        "pageInfo": Object {
          "__typename": "PageInfo",
          "endCursor": "Y3Vyc29yOnYyOpK5MjAyMy0wMi0wOFQxMToyMTozMCswMTowMM5Ux4u7",
          "hasNextPage": true,
        },
        "totalCount": 5,
      },
      "createdAt": "2023-02-08T08:50:14Z",
      "id": "I_kwDOAJy2Ks5d65oG",
      "number": 26126,
      "state": "CLOSED",
      "stateReason": "NOT_PLANNED",
      "title": "Bug: Failed when set \\"homepage\\" in package.json or set PUBLIC_URL",
      "url": "https://github.com/facebook/react/issues/26126",
    },
  },
}
`;

exports[`issue-hooks useIssues should load issues: query 1`] = `
Array [
  "https://api.github.com/graphql",
  Object {
    "body": "{\\"operationName\\":\\"getIssues\\",\\"variables\\":{\\"cursor\\":null,\\"search\\":\\"repo:facebook/react type:issue in:title test\\"},\\"query\\":\\"query getIssues($cursor: String, $search: String!) {\\\\n  search(query: $search, after: $cursor, type: ISSUE, first: 10) {\\\\n    issueCount\\\\n    nodes {\\\\n      ... on Issue {\\\\n        id\\\\n        createdAt\\\\n        title\\\\n        state\\\\n        number\\\\n        url\\\\n        author {\\\\n          login\\\\n          __typename\\\\n        }\\\\n        __typename\\\\n      }\\\\n      __typename\\\\n    }\\\\n    pageInfo {\\\\n      hasNextPage\\\\n      endCursor\\\\n      __typename\\\\n    }\\\\n    __typename\\\\n  }\\\\n}\\"}",
    "headers": Object {
      "accept": "*/*",
      "content-type": "application/json",
    },
    "method": "POST",
    "signal": AbortSignal {},
  },
]
`;

exports[`issue-hooks useIssues should load issues: query result 1`] = `
Object {
  "search": Object {
    "__typename": "SearchResultItemConnection",
    "issueCount": 12030,
    "nodes": Array [
      Object {
        "__typename": "Issue",
        "author": Object {
          "__typename": "User",
          "login": "weidehai",
        },
        "createdAt": "2023-02-12T05:31:50Z",
        "id": "I_kwDOAJy2Ks5ePiI3",
        "number": 26156,
        "state": "CLOSED",
        "title": "Bug: build args parse error on powsershell",
        "url": "https://github.com/facebook/react/issues/26156",
      },
      Object {
        "__typename": "Issue",
        "author": Object {
          "__typename": "User",
          "login": "Yi-19",
        },
        "createdAt": "2023-02-10T17:36:03Z",
        "id": "I_kwDOAJy2Ks5eLh-t",
        "number": 26149,
        "state": "OPEN",
        "title": "Unable to establish connection with the sandpack bundler.[DevTools Bug]: ",
        "url": "https://github.com/facebook/react/issues/26149",
      },
    ],
    "pageInfo": Object {
      "__typename": "PageInfo",
      "endCursor": "Y3Vyc29yOjI=",
      "hasNextPage": true,
    },
  },
}
`;

exports[`issue-hooks useIssues should load more issues: query 1`] = `
Array [
  "https://api.github.com/graphql",
  Object {
    "body": "{\\"operationName\\":\\"getIssues\\",\\"variables\\":{\\"cursor\\":\\"Y3Vyc29yOjI=\\",\\"search\\":\\"repo:facebook/react type:issue in:title test\\"},\\"query\\":\\"query getIssues($cursor: String, $search: String!) {\\\\n  search(query: $search, after: $cursor, type: ISSUE, first: 10) {\\\\n    issueCount\\\\n    nodes {\\\\n      ... on Issue {\\\\n        id\\\\n        createdAt\\\\n        title\\\\n        state\\\\n        number\\\\n        url\\\\n        author {\\\\n          login\\\\n          __typename\\\\n        }\\\\n        __typename\\\\n      }\\\\n      __typename\\\\n    }\\\\n    pageInfo {\\\\n      hasNextPage\\\\n      endCursor\\\\n      __typename\\\\n    }\\\\n    __typename\\\\n  }\\\\n}\\"}",
    "headers": Object {
      "accept": "*/*",
      "content-type": "application/json",
    },
    "method": "POST",
    "signal": AbortSignal {},
  },
]
`;

exports[`issue-hooks useIssues should load more issues: query result 1`] = `
Object {
  "search": Object {
    "__typename": "SearchResultItemConnection",
    "issueCount": 12030,
    "nodes": Array [
      Object {
        "__typename": "Issue",
        "author": Object {
          "__typename": "User",
          "login": "weidehai",
        },
        "createdAt": "2023-02-12T05:31:50Z",
        "id": "I_kwDOAJy2Ks5ePiI3",
        "number": 26156,
        "state": "CLOSED",
        "title": "Bug: build args parse error on powsershell",
        "url": "https://github.com/facebook/react/issues/26156",
      },
      Object {
        "__typename": "Issue",
        "author": Object {
          "__typename": "User",
          "login": "Yi-19",
        },
        "createdAt": "2023-02-10T17:36:03Z",
        "id": "I_kwDOAJy2Ks5eLh-t",
        "number": 26149,
        "state": "OPEN",
        "title": "Unable to establish connection with the sandpack bundler.[DevTools Bug]: ",
        "url": "https://github.com/facebook/react/issues/26149",
      },
      Object {
        "__typename": "Issue",
        "author": Object {
          "__typename": "User",
          "login": "naseer036",
        },
        "createdAt": "2023-02-10T11:33:26Z",
        "id": "I_kwDOAJy2Ks5eJY9I",
        "number": 26147,
        "state": "OPEN",
        "title": "Bug: react context not being provided to the components rendered that was passed as a prop.",
        "url": "https://github.com/facebook/react/issues/26147",
      },
      Object {
        "__typename": "Issue",
        "author": Object {
          "__typename": "User",
          "login": "catchex",
        },
        "createdAt": "2023-02-10T08:23:57Z",
        "id": "I_kwDOAJy2Ks5eIUJ8",
        "number": 26146,
        "state": "OPEN",
        "title": "Bug: Why is ref.current null when I get it in the useFetch callback?",
        "url": "https://github.com/facebook/react/issues/26146",
      },
    ],
    "pageInfo": Object {
      "__typename": "PageInfo",
      "endCursor": "Y3Vyc29yOjQ=",
      "hasNextPage": true,
    },
  },
}
`;
